### PR TITLE
Behat tests for REST Views

### PR DESCRIPTION
**Pull request by bdunogier**
> Story: https://jira.ez.no/browse/EZP-24773
> Status: WIP, behat scenario works

Implements a [Behat Scenario for REST /views](https://github.com/ezsystems/ezpublish-kernel/blob/ezp24671-rest_views_bdd/eZ/Bundle/EzPublishRestBundle/Features/Views/query.feature) (only ContentQuery for now).

### TODO
- [x] Commit matching code for BehatBundle
- [x] Commit matching code for ezplatform (if any)
- [x] Fix CS, it seems...

### CODE REVIEW

**Comment by bdunogier** [comment]
Unit tests fail because of the base branch, but behat seems fine. See https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/78796099, search for "@views".

**Comment by bdunogier** [comment]
Ping @miguelcleverti & @rpmsauron.

**Comment by andrerom** [comment]
So behat tests uses REST client? vaguely rings a bell, but then that also means there is value in also having the other Functional tests as it is able to test the actual http responses.

**Comment by bdunogier** [comment]
> So behat tests uses REST client?

TL;DR: Yes, they do. It was what was decided when we got started on Behat. It provided direct answers to challenges the team was having (like parsing and serializing values).

It actually uses components of it, the same than the REST Server: JSON & XML Input Handlers, Value Object Visitors and Input Parsers can be used to talk to the API. We only have a part of them at our disposal, but it saves some time already. Part of the work will be to actually write those handlers (48389c7).

It will have to use the full client at some point, as more complex handling (like FieldTypes processing) requires the repository's service.

The side benefit is that those implementations will be directly usable by the REST client.

> that also means there is value in also having the other Functional tests as it is able to test the actual http responses
I'm not sure how it'd really differ. The Behat tests have access to the HTTP client, and can test everything the other tests can do. Furthermore, maintaining both would be a lot of work (parsing of responses is purely manual, and only in JSON iirc, in those.

But there would be value, until we have proper BDD coverage, into getting them to run on travis with the failing ones skipped. And instead of fixing them, we should implement them, over time, on behat.

One (very big) imho benefit is that by doing this, in addition to having human readable tests, all of the visitors and parsers are bricks for the REST Client.

**Comment by andrerom** [comment]
> The side benefit is that those implementations will be directly usable by the REST client.

:+1: 

> But there would be value, until we have proper BDD coverage, into getting them to run on travis with the failing ones skipped. And instead of fixing them, we should implement them, over time, on behat.

:+1: 


Ok, ar as a service is all green as well, as opposed to red, grey or yellow ;)

**Comment by bdunogier** [comment]
See #1400 for functional tests.

**Comment by miguelcleverti** [comment]
looks good to me :+1:

**Comment by bdunogier** [comment]
Base branch merged, rebasing.

File: bin/.travis/prepare_behat.sh [hunk 1]
```diff
@@ -13,4 +13,4 @@ git clone --depth 1 --single-branch --branch master https://github.com/ezsystems
 cd ezplatform
 
 # Install everything needed for behat testing, using our local branch of this repo
-./bin/.travis/setup_from_external_repo.sh $BRANCH_BUILD_DIR "ezsystems/ezpublish-kernel:dev-tmp_travis_branch as 6.0"
+./bin/.travis/setup_from_external_repo.sh $BRANCH_BUILD_DIR "ezsystems/ezpublish-kernel:dev-ezp24671-rest_views_bdd as 6.0"
```
Comment (andrerom) [inline bin/.travis/prepare_behat.sh]: note: remember to fix back :)

**Comment by andrerom** [comment]
+1
